src/plugins/links/common/content_management/v1/cm_services.ts

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { schema } from '@kbn/config-schema';
+import type { ContentManagementServicesDefinition as ServicesDefinition } from '@kbn/object-versioning';
+import {
+  savedObjectSchema,
+  createResultSchema,
+  updateOptionsSchema,
+  createOptionsSchemas,
+  objectTypeToGetResultSchema,
+} from '@kbn/content-management-utils';
+import { DASHBOARD_LINK_TYPE, EXTERNAL_LINK_TYPE } from '.';
+import { LINKS_HORIZONTAL_LAYOUT, LINKS_VERTICAL_LAYOUT } from './constants';
+
+const baseLinkSchema = {
+  id: schema.string(),
+  label: schema.maybe(schema.string()),
+  order: schema.number(),
+};
+
+const dashboardLinkSchema = schema.object({
```

Comment:
Since these schemas are in common. I'd double check that those aren't important on the client by mistake. If I am not mistaken `schema` can pull a lot of unnecessary js to the client

Reply:
I believe we've handled this here.

https://github.com/elastic/kibana/blob/navigation-embeddable/src/plugins/links/common/content_management/cm_services.ts#L14-L17